### DISCUSSION TITLE
Allow adding click commands to typer apps for unified UX

### DISCUSSION THREAD

**Discussion by dionhaefner** [fastapi/typer#920] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer()

@app.callback()
def callback():
    pass

@app.command()
def hello(name: str):
    typer.echo(f"Hello {name}")

click_app = typer.main.get_command(app)

@click_app.command()
def goodbye(name: str):
    typer.echo(f"Goodbye {name}")

if __name__ == "__main__":
    click_app()
```

### Description

I am using click to define some commands dynamically at runtime and would like to add them to typer in a way that provides a uniform UX across the app.

The above code works fine but leads to differently formatted help messages for typer vs. click commands:

<img width="822" alt="Screenshot 2024-08-12 at 11 37 54" src="https://github.com/user-attachments/assets/0d62dcd1-fec7-4611-b566-6376046205f9">

Is there any way to achieve this?

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

Python 3.10.13

**Comment by daneah** [comment]
I'm interested in something I feel is related to this:

My use case is building a Typer-based CLI that composes a multitude of different tools, including a plugin-based architecture so that different teams can have different features. One thing I've wanted to be able to do is compose in a third-party existing Click app dynamically so that I can offer an easy way for folks to find useful tools and use them within the same CLI ecosystem, instead of directing them to install them as separate standalone tools. The existing approach in the docs doesn't quite work for this, because the Click app isn't composed up at the root `Typer` instance where I could invoke the resulting `.get_command()` object.

It would be great to have a way to "upgrade" a Click app into a Typer app, even if it doesn't have all the same niceties (e.g. because the Click app isn't type hinted inherently, I imagine some kinds of things would be difficult or impossible to infer).

I wanted to do this with an existing Click app I had, but ultimately went the route of converting it to Typer instead, which allowed me to integrate it into this central CLI project.
